app/src/main/java/com/example/myapplication/presentation/ui/SearchFragment.kt

```diff
@@ -0,0 +1,216 @@
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        _binding = FragmentSearchBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onStart() {
+        super.onStart()
+        adapter = CityAdapter {
+            startDetailInformationFragment(it)
+            //TODO()
+            //searchPresenter.clickOnRecyclerItem(it)
+        }
+        with(binding){
+            rvNearestCities.adapter = adapter
+            rvNearestCities.addItemDecoration(DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL))
+            swipeRefreshLayout.setOnRefreshListener{
+                requestPermissions()
+                swipeRefreshLayout.isRefreshing = false
+                //TODO()
+            }
+        }
+        requestPermissions()
+
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        _binding = null
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu, menu)
+        super.onCreateOptionsMenu(menu, inflater)
+        searchView = menu.findItem(R.id.action_search)?.actionView as androidx.appcompat.widget.SearchView
+        searchView.setOnQueryTextListener(object :
+            androidx.appcompat.widget.SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(p0: String?): Boolean {
+                findCityByName(p0 ?: "")
+                return true
+            }
+            override fun onQueryTextChange(p0: String?): Boolean {
+                return true
+            }
+
+        })
+    }
+
+    override fun onRequestPermissionsResult(
+            requestCode: Int,
+            permissions: Array<out String>,
+            grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == PERMISSION_GEO) {
+            var areGranted = true
+            grantResults.forEach {
+                if (it == PackageManager.PERMISSION_DENIED) {
+                    areGranted = false
+                }
+            }
+            if (areGranted) {
+                Log.d("MYTAG", "Получены пермишены")
+                getGeo()
+            } else {
+                showCitiesWeatherByDefault()
+            }
+        }
+    }
+//
+//    fun initPresenter(){
+//        searchPresenter = SearchPresenter(this, FindCityUseCase(WeatherRepositoryImpl(ApiFactory.weatherApi,
+//            WeatherDatabase.getInstance(requireContext()).weatherDAO),  Dispatchers.IO))
+//    }
+
+    private fun findCityByName(name: String) {
+        lifecycleScope.launch {
+            try {
+                val weather = findCityUseCase.getWeatherByCityName(name)
+                weather?.let {
```

Comment:
weather?.let {
 startDetailInformationFragment(it.id)
 } ?: run {
 showSnackBar(getString(R.string.city_not_found))
 }